kubernetes/client/ws_client.py

```diff
@@ -203,18 +202,21 @@ def close(self, **kwargs):
 WSResponse = collections.namedtuple('WSResponse', ['data'])
 
 
+def get_websocket_url(url):
+    parsed_url = urlparse(url)
+    parts = list(parsed_url)
+    if parsed_url.scheme == 'http':
+        parts[0] = 'ws'
+    else:
+        parts[0] = 'wss'
+    return urlunparse(parts)
+
+
 def websocket_call(configuration, url, query_params, _request_timeout,
                    _preload_content, headers):
     """An internal function to be called in api-client when a websocket
     connection is required."""
 
-    # switch protocols from http to websocket
-    url = url.replace('http://', 'ws://')
-    url = url.replace('https://', 'wss://')
-
-    # patch extra /
-    url = url.replace('//api', '/api')
```

Comment:
No, it's not present on the new code, but I don't see why it should be required. 

Do you have some sample url that uses this replace?